# This patch is DEPRECATED. Use fixInvokeBrowser.patch instead. That one will
# take care of all invokeBrowser() calls in KDE 4 apps and libraries, not just
# the one in KBugReport (used in the Help->Report Bug... menu item).
#
diff --git a/kdeui/dialogs/kbugreport.cpp b/kdeui/dialogs/kbugreport.cpp
index f9bc296..23b5863 100644
--- a/kdeui/dialogs/kbugreport.cpp
+++ b/kdeui/dialogs/kbugreport.cpp
@@ -27,9 +27,14 @@
 #include <QtGui/QGroupBox>
 #include <QtGui/QCloseEvent>
 
+#ifdef Q_OS_MAC
+#include <QDesktopServices>
+#else
+#include <ktoolinvocation.h>
+#endif
+
 #include <kaboutdata.h>
 #include <kcombobox.h>
-#include <ktoolinvocation.h>
 #include <kdebug.h>
 #include <klineedit.h>
 #include <klocale.h>
@@ -413,7 +418,16 @@ void KBugReportPrivate::_k_slotSetFrom()
 void KBugReport::accept()
 {
     if ( d->submitBugWeb ) {
+#ifdef Q_OS_MAC
+        // KToolInvocation::invokeBrowser() does not work easily on Apple OS X
+        // and QDesktopServices::openUrl() automatically gets you the user's
+        // preferred browser, as configured in Apple OS X preferences.  Maybe
+        // QDesktopServices should be used on all desktops, being more portable
+       // Note: d->url.url() is of type QString, which is correct for both.
+        QDesktopServices::openUrl( d->url.url() );
+#else
         KToolInvocation::invokeBrowser( d->url.url() );
+#endif
         return;
     }
 
